rudder-core/src/main/scala/com/normation/rudder/services/policies/nodeconfig/NodeConfigurationCacheRepository.scala

```diff
@@ -52,86 +52,137 @@ import net.liftweb.common.Loggable
   , cacheValue: Int
 )
 
-case class NodeConfigurationCache(
+/**
+ * NodeConfigurationHash keep a track of anything that would mean
+ * the node policies have changed and should be rewritten.
+ * We are at the node granularity check, because today, we don't
+ * know how to rewrite only some directives.
+ *
+ * Keep in mind that anything that is changing the related resources
+ * of policies that are written for the node during policy generation
+ * must be taken into account in the hash.
+ * Typically, the technique non-template resources, like pure CFEngine
+ * file or other configuration files, must be looked for change
+ * (for them, this is done with the technique "acceptation" (i.e commit)
+ * date.
+ *
+ */
+case class NodeConfigurationHash(
     id: NodeId
   , writtenDate: Option[DateTime]
-  , nodeInfoCache: Int
-  , parameterCache: Int
-  , nodeContextCache: Int
-  , policyCache: Set[PolicyCache]
-
+  , nodeInfoHash: Int
+  , parameterHash: Int
+  , nodeContextHash: Int
+  , policyHash: Set[policyHash]
 ) {
-  // We need a method to correctly compare a NodeConfigurationCache that was serialized
-  // from a NodeconfigurationCache created from a NodeConfiguration (so without writtenDate)
-  def equalWithoutWrittenDate(other: NodeConfigurationCache) : Boolean = {
+  // We need a method to correctly compare a NodeConfigurationHash that was serialized
+  // from a NodeConfigurationHash created from a NodeConfiguration (so without writtenDate)
+  def equalWithoutWrittenDate(other: NodeConfigurationHash) : Boolean = {
     id               == other.id &&
-    nodeInfoCache    == other.nodeInfoCache &&
-    parameterCache   == other.parameterCache &&
-    nodeContextCache == other.nodeContextCache &&
-    policyCache      == other.policyCache
+    nodeInfoHash    == other.nodeInfoHash &&
+    parameterHash   == other.parameterHash &&
+    nodeContextHash == other.nodeContextHash &&
+    policyHash      == other.policyHash
   }
 }
 
-/*
- * That object should be a service, no ?
- */
-object NodeConfigurationCache {
+object NodeConfigurationHash {
 
-  def apply(nodeConfig: NodeConfiguration): NodeConfigurationCache = {
+  /**
+   * Build the hash from a node configuration.
+   *
+   */
+  def apply(nodeConfig: NodeConfiguration): NodeConfigurationHash = {
 
 
     /*
-     * Compute the hash cash value for a set of variables:
-     * - the order does not matter
-     * - duplicate (same name) variable make no sense, so we remove them
-     * - only the ORDERED values matters
+     * A parameter update must lead to a regeneration of the node,
+     * because it could be used in pur CFEngine (when used in
+     * directive, we already tracked it with directives values).
+     *
+     * So a modification in ANY parameter lead to a regeneration
+     * of ALL NODES (everything, yes).
      *
-     * Also, variable with no values are equivalent to no variable, so we
-     * remove them.
+     * For parameters, do not consider:
```

Comment:
don't you mean properties ? as parameters are overridable, but there is nothing to override them